graphiql-spring-boot-autoconfigure/src/main/resources/graphiql.html

```diff
@@ -162,8 +162,12 @@
   props.onEditVariables = onEditVariables
   props.onEditOperationName = onEditOperationName
   props.onEditHeaders = onEditHeaders
+  props.headers = props.headers || {}
+  if (headers) {
+    var newHeaders = Object.assign({}, JSON.parse(props.headers), headers)
+    props.headers = JSON.stringify(newHeaders, undefined, 2)
+  }
 
```

Comment:
calling `onEditHeaders(props.headers)` seems to be necessary here, otherwise the headers are not sent with the request (until the `onEditHeaders `event is triggered)

Also, two additonal headers are appearing on the "Request headers" tab: `"Accept": "application/json"` and `"Content-Type": "application/json"`

Reply:
Those two additional headers are added by the `GraphiQLController` atm because otherwise it would send the request as just plain text:

Accept: */*
Content-Type: text/plain;charset=UTF-8

According to [GraphQL Specification](https://graphql.org/learn/serving-over-http/#post-request) it should use Content-Type application/json in that case.

I could add them "silently" though, so instead of adding them in the headers tab too, only add them if there absent.

Reply:
I think the "silent" solution is better in this case, since these headers are/should be always the same. (And the user still has the option to customize them by adding them manually.)